tests/next_tests/unit_tests/embedded_tests/test_nd_array_field.py

```diff
@@ -125,3 +131,54 @@ def fma(a: common.Field, b: common.Field, c: common.Field, /) -> common.Field:
 
     result = fma(field_inp_a, field_inp_b, field_inp_c)
     assert np.allclose(result.ndarray, expected)
+
+
+def test_get_slices_with_named_indices_1d_to_2d_missing_dim_right():
+    field_domain = common.Domain(dims=(IDim,), ranges=(UnitRange(0, 10),))
+    new_domain = common.Domain(dims=(IDim, JDim), ranges=(UnitRange(5, 10), UnitRange(5, 10)))
+    field = common.field(np.ones((10,)), domain=field_domain)
+    slices = _get_slices_with_named_indices(field, new_domain)
+    assert slices == (slice(5, 10, None), None)
+
+
+def test_get_slices_with_named_indices_1d_to_2d_missing_dim_left():
+    field_domain = common.Domain(dims=(JDim,), ranges=(UnitRange(0, 10),))
+    new_domain = common.Domain(dims=(IDim, JDim), ranges=(UnitRange(0, 5), UnitRange(0, 10)))
+    field = common.field(np.ones((10,)), domain=field_domain)
+    slices = _get_slices_with_named_indices(field, new_domain)
+    assert slices == (None, slice(0, 10, None))
+
+
+def test_get_slices_with_named_indices_1d_to_3d():
+    field_domain = common.Domain(dims=(IDim,), ranges=(UnitRange(0, 10),))
+    new_domain = common.Domain(dims=(IDim, JDim, KDim), ranges=(UnitRange(0, 5), UnitRange(0, 10), UnitRange(0, 10)))
+    field = common.field(np.ones((10,)), domain=field_domain)
+    slices = _get_slices_with_named_indices(field, new_domain)
+    assert slices == (slice(0, 5, None), None, None)
+
+
+def test_get_slices_with_named_indices_3d_to_1d():
+    field_domain = common.Domain(dims=(IDim, JDim, KDim), ranges=(UnitRange(0, 10), UnitRange(0, 10), UnitRange(0, 10)))
+    new_domain = common.Domain(dims=(IDim,), ranges=(UnitRange(0, 10),))
+    field = common.field(np.ones((10, 10, 10)), domain=field_domain)
+    slices = _get_slices_with_named_indices(field, new_domain)
+    assert slices == (slice(0, 10, None),)
+
```

Comment:
test cases for NamedIndex missing, something like:
```suggestion
def test_get_slices_with_named_indices_and_ranges:
    field_domain = common.Domain(dims=(IDim, JDim, KDim), ranges=(UnitRange(0, 10), UnitRange(0, 10), UnitRange(0, 10)))
    new_domain = ((IDim,UnitRange(0, 10),(IDim, 2), (KDim, 3)) # <--- indices used
    field = common.field(np.ones((10, 10, 10)), domain=field_domain)
    slices = _get_slices_with_named_indices(field, new_domain)
    assert slices == (slice(0, 10, None),)
```